### PR TITLE
Fix txhash encoding

### PR DESCRIPTION
**Pull request by ssd04**

### CODE REVIEW

File: data/transaction.go [hunk 1]
```diff
@@ -118,24 +118,25 @@ type TransactionsPool struct {
 
 // TransactionWithOrder defines a wrapper over transaction
 type TransactionWithOrder struct {
```
Comment (iulianpascalau) [inline data/transaction.go:120]: TransactionWithFeesAndOrder ?

Comment (ssd04) [reply data/transaction.go:120]: I've added feeInfo only for some testing, forgot to remove it, will handle FeeInfo in a separate PR

Comment (ssd04) [reply data/transaction.go:120]: removed feeInfo field

File: data/transaction.go [hunk 2]
```diff
@@ -118,24 +118,24 @@ type TransactionsPool struct {
...
-	transaction.Transaction
-	ExecutionOrder int
+	TransactionHandler *transaction.Transaction
+	ExecutionOrder     int
```
Comment (mariusmihaic) [inline data/transaction.go:122]: [comment]

Could've made this field uint32

File: process/eventsInterceptor.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ func (ei *eventsInterceptor) getLogEventsFromTransactionsPool(logs []*data.LogDa
 		for _, eventHandler := range logData.LogHandler.GetLogEvents() {
 			le := &data.LogEvent{
 				EventHandler: eventHandler,
-				TxHash:       hex.EncodeToString([]byte(logData.TxHash)),
+				TxHash:       logData.TxHash,
```
Comment (mariusmihaic) [inline process/eventsInterceptor.go:75]: [question]

So this is a hex encoded string now?

**Review by mariusmihaic — APPROVED** [review]
Would've used `uint32` for `ExecutionOrder`, but that's my preference.
I've approved it anyway, in case you strongly disagree with it, you can ignore it

**Review by mariusmihaic — COMMENTED** [review]
I also use a similar proto message for firehose in mx-chain-core
```
message TxInfo {
  Transaction Transaction = 1;
  FeeInfo     FeeInfo = 2;
  uint32      ExecutionOrder = 3;
}
```

Not sure if useful for you in the future

File: process/eventsInterceptor.go (hunk 3 above)
Comment (ssd04) [reply process/eventsInterceptor.go:75]: yes, it is encoded on the sender side

**Comment by ssd04** [comment]
> I also use a similar proto message for firehose in mx-chain-core
> 
> ```
> message TxInfo {
>   Transaction Transaction = 1;
>   FeeInfo     FeeInfo = 2;
>   uint32      ExecutionOrder = 3;
> }
> ```
> 
> Not sure if useful for you in the future

yes, maybe we can evaluate adding that in core, in outport

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**